lib/player.js

```diff
@@ -3891,9 +3939,30 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
 
     this.manifest_.textStreams.push(stream);
     this.onTracksChanged_();
-    return shaka.util.StreamUtils.textStreamToTrack(stream);
+    const track = shaka.util.StreamUtils.textStreamToTrack(stream);
+    return Promise.resolve(track);
+  }
+
+
+  /**
+   * @param {string} uri
+   * @param {!shaka.net.NetworkingEngine} netEngine
+   * @param {shaka.extern.RetryParameters} retryParams
+   * @return {!Promise.<string>}
+   * @private
+   */
+  async getTextData_(uri, netEngine, retryParams) {
+    const type = shaka.net.NetworkingEngine.RequestType.MANIFEST;
```

Comment:
It's a minor nit, but a subtitle file should be considered a text segment for the purposes of filtering, so this should be set to `SEGMENT` instead of `MANIFEST`.